src/main/java/org/wildfly/security/auth/server/event/RealmIdentityTimeoutUpdateEvent.java

```diff
@@ -0,0 +1,65 @@
+/*
+ * JBoss, Home of Professional Open Source.
+ * Copyright 2016 Red Hat, Inc., and individual contributors
+ * as indicated by the @author tags.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.wildfly.security.auth.server.event;
+
+import org.wildfly.security.auth.server.RealmIdentity;
+import org.wildfly.security.auth.server.RealmUnavailableException;
+
+/**
+ * An event indicating a timeout attribute change for a realm identity.
+ *
+ * @author <a href="mailto:fjuma@redhat.com">Farah Juma</a>
+ */
+public final class RealmIdentityTimeoutUpdateEvent extends RealmEvent {
```

Comment:
This seems similar to the credential event problem.  But it's a little different because the timeout is pretty mechanism-specific, at least the way we utilize it.  I have some tangentially related-to-timeout things that I'd like to discuss but I don't think they're directly relevant to the PR so I'll bring it up in chat sometime soon.

I don't really have a good idea for a solution to this at the moment.